gemrb/core/PathFinder.cpp

```diff
@@ -133,7 +133,17 @@ bool Map::TargetUnreachable(const Point &s, const Point &d, unsigned int size, b
 {
 	int flags = PF_SIGHT;
 	if (actorsAreBlocking) flags |= PF_ACTORS_ARE_BLOCKING;
-	return FindPath(s, d, size, 0, flags) == nullptr;
+	PathNode *path = FindPath(s, d, size, 0, flags);
+	bool targetUnreachable = path == nullptr;
+	if (!targetUnreachable) {
+		PathNode *thisNode = path;
+		while (thisNode) {
+			PathNode *nextNode = thisNode->Next;
+			delete(thisNode);
+			thisNode = nextNode;
+		}
```

Comment:
This would fit nicely in a destructor. Reading up on c++11 POD changes, it would even still keep PathNode POD, since it would satisfy "standard layout" — if I understand correctly.

Reply:
I guess it would have to be selective though, since most of the time we don't want to wipe everything that's linked. :)

Reply:
I guess again, if the idea is feasible at all, it would better fit in a standalone method and then there's no fallout.

Reply:
We already have `Movable::ClearPath`, which is where I copied this from. There are no other uses of PathNode than for Movables iirc and even then I'd rather just get a move on and replace PathNode with an `std::list`

Reply:
ok